src/tests/nativeaot/SmokeTests/HardwareIntrinsics/Program.cs

```diff
@@ -50,100 +50,208 @@ static int Main()
         //
         // The test is compiled with multiple defines to test this.
 
+        bool ExpectedVectorsAccelerated = true;
```

Comment:
Are these changes just adding more coverage, or are there also a functional changes from before (ie the existing tests fails with the runtime changes in this PR)?

Reply:
This is adding more test coverage, specifically over the ISAs that were missing (Avx512 + X86Serialize)